storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbInstance.java

```diff
@@ -154,16 +168,32 @@ public synchronized KvStoreTransaction startTransaction() {
     return createStream(column, setupIterator, key -> true);
   }
 
-  @SuppressWarnings("MustBeClosedChecker")
   @MustBeClosed
   private <K, V> Stream<ColumnEntry<K, V>> createStream(
       KvStoreColumn<K, V> column,
       Consumer<RocksIterator> setupIterator,
       Predicate<K> continueTest) {
+
+    return createStreamRaw(column, setupIterator, continueTest)
+        .map(
+            entry ->
+                ColumnEntry.create(
+                    column.getKeySerializer().deserialize(entry.getKey()),
+                    column.getValueSerializer().deserialize(entry.getValue())));
+  }
+
+  @SuppressWarnings("MustBeClosedChecker")
+  @MustBeClosed
+  private <K, V> Stream<ColumnEntry<byte[], byte[]>> createStreamRaw(
+      KvStoreColumn<K, V> column,
+      Consumer<RocksIterator> setupIterator,
+      Predicate<K> continueTest) {
     final ColumnFamilyHandle handle = columnHandles.get(column);
     final RocksIterator rocksDbIterator = db.newIterator(handle);
     setupIterator.accept(rocksDbIterator);
-    return RocksDbIterator.create(column, rocksDbIterator, continueTest, closed::get).toStream();
+    return RocksDbIterator.create(column, rocksDbIterator, continueTest, closed::get)
+        .toStream()
+        .onClose(rocksDbIterator::close);
```

Comment:
This is already done inside `RocksDbIterator.toStream()` so I don't think it's required here.